assets/js/ajax-filters.js

```diff
@@ -1,288 +1,458 @@
+					if ( window.console && 'abort' !== textStatus ) {
+						window.console.log( textStatus + ': ' + error );
+					}
+				},
+				statusCode: {
+					404: function() {
+						if ( window.console ) {
+							window.console.log(
+								'Error 404: Ajax Endpoint cannot be reached. Go to Settings > Permalinks and save to resolve.'
+							);
+						}
+					},
+				},
+			} );
+		} );
+
+	$(
+		'#search_keywords, #search_location, .job_types :input, #search_categories, .job-manager-filter'
+	)
+		.change( function() {
+			var $target = $( this ).closest( 'div.job_listings' );
+			$target.triggerHandler( 'update_results', [ 1, false ] );
+			job_manager_store_state( $target );
+		} )
+		.on( 'keyup', function( e ) {
+			if ( e.which === 13 ) {
+				$( this ).trigger( 'change' );
+			}
+		} );
+
+	$( '.job_filters' )
+		.on( 'click', '.reset', function() {
+			var $target = $( this ).closest( 'div.job_listings' );
+			var $form = $( this ).closest( 'form' );
+
+			$form
+				.find(
+					':input[name="search_keywords"], :input[name="search_location"], .job-manager-filter'
+				)
+				.not( ':input[type="hidden"]' )
+				.val( '' )
+				.trigger( 'change.select2' );
+			$form
+				.find( ':input[name^="search_categories"]' )
+				.not( ':input[type="hidden"]' )
+				.val( '' )
+				.trigger( 'change.select2' );
+			$( ':input[name="filter_job_type[]"]', $form )
+				.not( ':input[type="hidden"]' )
+				.attr( 'checked', 'checked' );
+
+			$target.triggerHandler( 'reset' );
+			$target.triggerHandler( 'update_results', [ 1, false ] );
+			job_manager_store_state( $target );
+
+			return false;
+		} )
+		.on( 'submit', function() {
+			return false;
+		} );
+
+	$( document.body ).on( 'click', '.load_more_jobs', function() {
+		var $target = $( this ).closest( 'div.job_listings' );
+		var page = parseInt( $( this ).data( 'page' ) || 1, 10 );
+
+		$( this ).addClass( 'loading' );
+
+		page = page + 1;
+		$( this ).data( 'page', page );
+
+		$target.triggerHandler( 'update_results', [ page, true ] );
 		return false;
 	} );
 
 	if ( $.isFunction( $.fn.select2 ) && typeof job_manager_select2_args !== 'undefined' ) {
 		var select2_args = job_manager_select2_args;
-		select2_args['allowClear']              = true;
-		select2_args['minimumResultsForSearch'] = 10;
+		select2_args[ 'allowClear' ] = true;
+		select2_args[ 'minimum$resultsForSearch' ] = 10;
 
 		$( 'select[name^="search_categories"]:visible' ).select2( select2_args );
 	}
 
-	var $supports_html5_history = false;
-	if ( window.history && window.history.pushState ) {
-		$supports_html5_history = true;
-	}
+	// Initial job and $form population
+	$( window ).on( 'load', function() {
+		$( 'div.job_listings' ).each( function() {
+			var $target = $( this );
+			var $form = $target.find( '.job_filters' );
+			var $results_loaded = false;
+			var state = job_manager_get_state( $target );
+
+			if ( state ) {
+				// Restore the results from cache.
+				if (state.results) {
```

Comment:
Check spacing

Reply:
Done in 38b06765d730da12f993c82916a9ba82c9d3eb3f